library/catalog/static/js/bookDetails.js

```diff
@@ -1,49 +1,187 @@
+  $(".toast").toast({ delay: 3000 });
+  $(".toast").toast("show");
 
-$(document).ready(function(){
-    $('.toast').toast({delay: 3000});
-    $('.toast').toast('show');
+  let endpoint = document.getElementById("endpoint-list").getAttribute("url");
+  let csrfmiddlewaretoken = document.getElementsByName("csrfmiddlewaretoken")[0]
+    .value;
+
+  $.ajax({
+    method: "GET",
+    url: endpoint,
+    data: {
+      csrfmiddlewaretoken: csrfmiddlewaretoken,
+    },
+    dataType: "json",
+    success: fillCommentsList,
+  });
 });
 
-ratingStars.forEach(ratingStar => {
-    ratingStar.addEventListener('click', function () {
-        setWithExpiry(starStorageKey, parseInt(ratingStar.value), timeToLiveMin);
-    })
-})
+ratingStars.forEach((ratingStar) => {
+  ratingStar.addEventListener("click", function () {
+    setWithExpiry(starStorageKey, parseInt(ratingStar.value), timeToLiveMin);
+  });
+});
 
 function setWithExpiry(key, value, ttl) {
-    let now = new Date()
-    let item = {
-        value: value,
-        expiry: now.getTime() + ttl * 60000,
-    }
-    localStorage.setItem(key, JSON.stringify(item));
+  let now = new Date();
+  let item = {
+    value: value,
+    expiry: now.getTime() + ttl * 60000,
+  };
+  localStorage.setItem(key, JSON.stringify(item));
 }
 
 function getWithExpiry(key) {
-    const itemStr = localStorage.getItem(key);
-    if (!itemStr) {
-        return null;
-    }
-    let item = JSON.parse(itemStr);
-    let now = new Date();
-    if (now.getTime() > item.expiry) {
-        localStorage.removeItem(key);
-        return null;
-    }
-    return item.value
+  const itemStr = localStorage.getItem(key);
+  if (!itemStr) {
+    return null;
+  }
+  let item = JSON.parse(itemStr);
+  let now = new Date();
+  if (now.getTime() > item.expiry) {
+    localStorage.removeItem(key);
+    return null;
+  }
+  return item.value;
 }
 
 function saveStarRating(url) {
-    let starValue = getWithExpiry(starStorageKey);
-    if (starValue) {
-        url = url + `${starValue}/`;
+  let starValue = getWithExpiry(starStorageKey);
+  if (starValue) {
+    url = url + `${starValue}/`;
+  }
+  let request = new XMLHttpRequest();
+  request.open("GET", url, true);
+  request.send(null);
+}
+
+function addMouseEventsForComments() {
+  let commentArea = document.querySelectorAll("#commentArea");
+  commentArea.forEach((comment) => {
+    comment.addEventListener("mouseover", function (event) {
+      comment.querySelector("#commentButton").style.display = "block";
+    });
+    comment.addEventListener("mouseout", function (event) {
+      comment.querySelector("#commentButton").style.display = "none";
+    });
+  });
+}
+
+function fillCommentsList(list) {
+  let userId = document.getElementById("userId").getAttribute("value");
+  let userRole = document.getElementById("userRole").getAttribute("value");
+
+  const data = list.reviews;
+  $("#commentList").empty();
+
+  for (let i = 0; i < data.length; i++) {
```

Comment:
Is it the best way to add buttons and comments ?
Is it possible to do this on template level ?

Reply:
no
if we use ajax we must add a button only in this way

> Is it the best way to add buttons and comments ?
> Is it possible to do this on template level ?

no
if we use ajax we must add a button only in this way